src/test/java/step3/domain/car/LocationTest.java

```diff
@@ -0,0 +1,60 @@
+package step3.domain.car;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.NullSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+
+class LocationTest {
+
+    private static final int LOCATION = 0;
+    private static final int INTERVAL = 2;
+
+    @Test
+    void createLocation_nullInput_thrownException() {
+        assertThatIllegalArgumentException()
+                .isThrownBy(() -> Location.placeOn(null))
+                .withMessage("location is required");
+    }
+
+
+    @ParameterizedTest(name = "[{index}] interval: {0}")
+    @NullSource
+    @CsvSource(value = {"-1"})
+    void createLocation_NotPositiveInterval_thrownException(Integer interval) {
+        assertThatIllegalArgumentException()
+                .isThrownBy(() -> Location.placeOn(LOCATION, interval))
+                .withMessage("interval is not Positive");
+    }
+
+    @DisplayName("전진하면 거리가 interval 만큼 증가한다.")
+    @Test
+    void goForwardTest() {
+        //given
+        Location location = Location.placeOn(LOCATION, INTERVAL);
+
+        //when
+        location.goForward();
+
+        //then
+        assertThat(location).isEqualTo(Location.placeOn(LOCATION + INTERVAL));
```

Comment:
```suggestion
        //given
        Location location = Location.placeOn(LOCATION);

        //when
        Location nextLocation = location.goForward(INTERVAL);

        //then
        assertThat(location).isEqualTo(nextLocation);
```

이런식으로 동작 할 수 있게 만들어보면 어떨까요?
Location을 goForward()의 반환값으로 써보면 좋을 것 같아요. :)

Reply:
goForward() 파라미터로 interval 이 꼭 필요할까요??
자동차의 매 순간 속력이 다르다면 파라미터로 받을 수 있겠지만
현재 요구사항으로는 Location 생성시점에 interval 을 정하는 것으로도 충분할 것 같아서요.
용대님은 어떻게 생각하시나요??

Reply:
음 이 부분은 기존의 설계를 따라가기 위해 제시한 방향이고
`Location 생성시점에 interval 을 정하는 것으로 충분하다`는 동혁님의 의견에 동의합니다!

Reply:
@yongdae
`기존의 설계를 따라가기 위해 제시한 방향이고` 가 의미하는게
placeOn() 과 goForward()의 반환 형식을 맞춘다는 뜻일까요??

placeOn() 정적팩토리메소드 이기 때문에 Location 객체를 반환하지만, 
goForward()가 Location객체를 반환하는게 어색한 것 같기도 해서 질문드려요!

Reply:
제가 제안한 코드가 기존의  설계를 따라가기 위해 제시한 반향이라는 의미였어요. :)

이야기하신 것처럼 goForward()는 어색한 부분이 있죠
그렇다면 메소드 이름을 next() 같은 이름을 사용해보면 어떨까요?